frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -468,9 +468,12 @@ func toDispatchState(ctx context.Context, dt []byte, opt ConvertOpt) (*dispatchS
 			if d.base == nil && !d.dispatched && !d.resolved {
 				d.resolved = reachable // avoid re-resolving if called again after onbuild
 				if d.stage.BaseName == emptyImageName {
-					d.state = llb.Scratch()
-					d.image = emptyImage(platformOpt.targetPlatform)
-					d.platform = &platformOpt.targetPlatform
+					platform := d.platform
+					if platform == nil {
+						platform = &platformOpt.targetPlatform
+					}
```

Comment:
Unsure if that's needed
```suggestion
					platform := d.platform
					if platform == nil {
						platform = &platformOpt.targetPlatform
					}
					d.platform = platform
```

Reply:
@tonistiigi could you please take a look at correctness here? I'm not so sure whether or not `d.platform` will be used later

Reply:
I don't think this is needed.

Reply:
Now I reviewed this again, it seems that `d.platform` was being set to the default target value by the code I removed. I should restore that I believe, but set to the chosen or default now